civis/ml/_model.py

```diff
@@ -450,3 +462,518 @@ def training_metadata(self):
+        try:
+            fut = ModelFuture(train_job_id, train_run_id, client)
+            container = client.scripts.get_containers(train_job_id)
+        except CivisAPIError as api_err:
+            if api_err.status_code == 404:
+                raise ValueError('There is no Civis Platform job with '
+                                 'script ID {} and run ID {}!'.format(
+                                     train_job_id, train_run_id)) from api_err
+            raise
+
+        args = container.arguments
+        model = args['MODEL']
+        dependent_variable = args['TARGET_COLUMN'].split()
+        primary_key = args.get('PRIMARY_KEY')
+        parameters = json.loads(args.get('PARAMS', {}))
+        cross_validation_parameters = json.loads(args.get('CVPARAMS', {}))
+        calibration = args.get('CALIBRATION')
+        excluded_columns = args.get('EXCLUDE_COLS', None)
+        if excluded_columns:
+            excluded_columns = excluded_columns.split()
+        cpu_requested = args.get('REQUIRED_CPU')
+        memory_requested = args.get('REQUIRED_MEMORY')
+        disk_requested = args.get('REQUIRED_DISK_SPACE')
+        name = container.name
+        if name.endswith(' Train'):
+            # Strip object-applied suffix
+            name = name[:-len(' Train')]
+
+        klass = cls(model=model,
+                    dependent_variable=dependent_variable,
+                    primary_key=primary_key,
+                    model_name=name,
+                    parameters=parameters,
+                    cross_validation_parameters=cross_validation_parameters,
+                    calibration=calibration,
+                    excluded_columns=excluded_columns,
+                    client=client,
+                    cpu_requested=cpu_requested,
+                    disk_requested=disk_requested,
+                    memory_requested=memory_requested,
+                    verbose=args.get('DEBUG', False))
+        klass.train_result_ = fut
+        return klass
+
+    def train(self, X=None, table_name=None, database_name=None, file_id=None,
+              sql_where=None, sql_limit=None, oos_scores=None,
+              oos_scores_db=None, if_exists='fail', fit_params=None,
+              polling_interval=None):
+        """Start a Civis Platform job to train your model
+
+        Provide input through one of
+        a local CSV or :class:`~pandas.DataFrame` (``X``),
+        a Civis Table (``table_name`` and ``database_name``), or
+        a Civis File containing a CSV (``file_id``).
+
+        Model outputs will always contain out-of-sample scores
+        (accessible through ``ModelFuture.table`` on this function's
+        output), and you may chose to store these out-of-sample scores
+        in a Civis Table with the ``oos_scores``, ``oos_scores_db``,
+        and ``if_exists`` parameters.
+
+        Parameters
+        ----------
+        X : pd.DataFrame or str, optional
```

Comment:
I would find the `train` and `train_from_local` split to be confusing. It would suggest to me that the two methods are doing something different, when I don't think they really are. I can divide `X` into `df` and `csv_path` if you think that would be clearer.